text/0000-libtest-json-output.md

```diff
@@ -0,0 +1,119 @@
+- Feature Name: libtest_json_output
+- Start Date: 2017-12-04
+- RFC PR:
+- Rust Issue:
+
+# Summary
+[summary]: #summary
+
+Add a machine-readable JSON-output mode for Rust's libtest.
+
+# Motivation
+[motivation]: #motivation
+
+Adding a machine-readable output for Rust's built-in tests will make external tool integration easier.
+
+Having this feature is not intended to replace the proposed custom test runners, but to enrich
+the default set of features offered by rust out-of-the-box.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Test binaries compiled `cargo test` or `rustc --test` allow the user to specify an output format using the `--format` flag.
+
+Using this flag, the format can be selected from one of the following:
+ - `pretty`: The default setting; print test results in a detailed manner.
+ - `terse`: Equivalent to `-q`; display one character per test instead of one line.
+ - `json`: Print to stdout the test result in json format. Each line is a complete JSON document describing one event.
+
+Each JSON object starts with the "type" property, which is one of the following:
+
+ - `suite`: Events regarding the whole testing suite; consists of a header specifying the amount of tests to be performed,
+ 	and a footer with results summary.
+ - `test`: A change in a test's state, along with its name. Can be one of the following:
```

Comment:
What is "its name" here? A unique identifier for the test? We have one of these, right? I assume it only needs to be unique in the current suite.

Reply:
Essentially it's the test's name+path.
I'm pretty sure they're unique, as Rust paths are mostly unique, but it requires a second check.